[fix] bookTag CRUD 수정

dudxor4587 opened the pull request:
# 설명
- url 매핑 수정
- 레퍼지토리에 코드 추가
- request 파일에 유효성 검증 추가
- response 파일에 private 추가
- service에서 예외처리 추가
- 예외처리 코드 추가

# 작업
- url 매핑을 규칙대로 변경하였습니다.
- 필요한 함수가 생겨 레포에 작성하였습니다.
- request 파일에 @valid 이용을 위한 유효성 검증 애너테이션을 추가하였습니다.
- response 파일에 private를 추가하였습니다.
- service 클래스에서 각 함수 별 예외처리 코드를 작성하였습니다.
 

(이슈에 대한 작업완료시) close #이슈번호

dudxor4587 committed:
[feature] BookTag CRUD 작성

dudxor4587 committed:
[fix] 파일 위치 및 500에러 수정

dudxor4587 committed:
Merge remote-tracking branch 'origin/feature/bookTag' into feature/bookTag

# Conflicts:
#	src/main/java/shop/bookbom/shop/domain/book/repository/BookRepository.java

dudxor4587 committed:
Merge remote-tracking branch 'origin/feature/bookTag' into feature/bookTag

# Conflicts:
#	src/main/java/shop/bookbom/shop/domain/book/repository/BookRepository.java

dudxor4587 committed:
[fix] 수정사항 반영
설명:
- url 매핑 수정
- 레퍼지토리에 코드 추가
- request 파일에 유효성 검증 추가
- response 파일에 private 추가
- service에서 예외처리 추가

작업:
- url 매핑을 규칙대로 변경하였습니다.
- 필요한 함수가 생겨 레포에 작성하였습니다.
- request 파일에 @valid 이용을 위한 유효성 검증 애너테이션을 추가하였습니다.
- response 파일에 private를 추가하였습니다.
- service 클래스에서 각 함수 별 예외처리 코드를 작성하였습니다.

dudxor4587 committed:
[fix] 예외처리 코드 추가

dudxor4587 committed:
[fix] merge 및 서비스 코드 수정

dudxor4587 committed:
[test] BookTag 테스트 코드 작성

minsu11 commented on src/main/java/shop/bookbom/shop/domain/booktag/service/BookTagService.java:
```diff
@@ -21,9 +25,13 @@ public class BookTagService {
     public List<BookTagResponse> getBookTagInformation(long bookId) {
         // 주어진 bookId에 해당하는 BookTag 목록을 가져옴
         List<BookTag> bookTags = bookTagRepository.findAllByBookId(bookId);
+        if (bookTags.isEmpty()) {
```
도서에 태그가 없는 경우에 exception을 터뜨리는 게 도서 요구 사항에 맞을까용??

dudxor4587 replied:
exception 대신 빈 리스트를 반환하도록 수정했습니다!

minsu11 commented on src/test/java/shop/bookbom/shop/domain/booktag/controller/BookTagControllerTest.java:
```diff
@@ -0,0 +1,79 @@
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.*;
+
+@WebMvcTest(BookTagController.class)
+class BookTagControllerTest {
```
유효성 검사 테스트도 해주세요!

dudxor4587 replied:
유효성 검사 테스트 추가했습니다!

minsu11 reviewed (commented):
수고하셨습니다!

hyeond0 commented on src/main/java/shop/bookbom/shop/domain/booktag/controller/BookTagController.java:
```diff
@@ -1,47 +1,50 @@
     // 책 태그 정보
-    @GetMapping("/shop/bookTag/{id}")
-    public ResponseEntity<Object> getBookTag(@PathVariable long id){
+    @GetMapping("/booktag/{id}")
```
/book/tag 는 어떨까요?

dudxor4587 replied:
태그 자체를 /book/tag로 수정하는게 더 괜찮을까요??

dudxor4587 replied:
우선 /boo/tag로 수정해서 커밋 했습니다!

hyeond0 commented on src/main/java/shop/bookbom/shop/domain/booktag/controller/BookTagController.java:
```diff
@@ -1,47 +1,50 @@
-    @PostMapping("/shop/bookTag")
-    public ResponseEntity<Integer> saveBookTag(@RequestBody BookTagRequest bookTagRequest){
+    @PostMapping("/booktag")
+    public CommonResponse<Void> saveBookTag(final @Valid @RequestBody BookTagRequest bookTagRequest) {
```
final을 다신 이유가 궁금합니다!

dudxor4587 replied:
요청받아온 값은 수정하지 못하게 하려고 final을 붙였습니다!

hyeond0 commented on src/main/java/shop/bookbom/shop/domain/booktag/dto/BookTagRequest.java:
```diff
@@ -1,9 +1,15 @@
 import lombok.Getter;
 
 @Getter
 public class BookTagRequest {
```
기본 생성자가 빠진 것 같아요!

hyeond0 commented on src/main/java/shop/bookbom/shop/domain/booktag/dto/BookTagResponse.java:
```diff
@@ -7,5 +7,5 @@
 @Setter
```
프로젝트 코드 컨벤션에 DTO 관련 부분 업데이트했습니다!
확인하고 반영해주시면 감사하겠습니다 :)

dudxor4587 replied:
DTO 관련 컨벤션 적용해서 수정사항 반영했습니다!

hyeond0 reviewed (approved):
고생하셨습니다!
궁금한 점 몇 가지 적어봤어요!

dudxor4587 committed:
[fix] 수정사항 반영
# 설명
- 도서에 태그가 없는 경우 빈 리스트 반환
- 유효성 검사 테스트 추가
- url 수정
- DTO 컨벤션 적용

# 작업
- 도서에 태그가 없는 경우 에러를 반환했는데, 빈 리스트를 반환하도록 수정
- 유효성 검증을 위한 테스트코드 추가
- url shop/booktag에서 shop/book/tag로 수정
- DTO컨벤션 적용하여 코드 수정

dudxor4587 committed:
[fix] test코드 url 수정

dudxor4587 committed:
Merge branch 'develop' into feature/bookTag

dudxor4587 committed:
[fix] get 요청 시 반환하는 tag 정보 추가

jinyesong reviewed (approved):
고생하셨습니다!

jinyesong commented on src/main/java/shop/bookbom/shop/domain/booktag/dto/request/BookTagCreateRequest.java:
```diff
@@ -0,0 +1,19 @@
+import lombok.NoArgsConstructor;
+
+@Getter
+@NoArgsConstructor
```
혹시 NoArgsConstructor가 어떤 경우에 사용되나요?
